pkg/reconciler/common/reconcile.go

```diff
@@ -253,3 +273,59 @@ func (r *GenericServiceReconciler) syncAdapterService(ctx context.Context,
 
 	return adapter, nil
 }
+
+// findAdapter returns the adapter object for a given source if it exists.
+func findAdapter(genericReconciler interface{}, src v1alpha1.EventSource) (metav1.Object, error) {
+	// the combination of standard labels {name,instance} is unique and
+	// immutable
+	sel := labels.SelectorFromSet(labels.Set{
+		AppNameLabel:     AdapterName(src),
+		AppInstanceLabel: src.GetName(),
+	})
+
+	var objs []metav1.Object
+	var gr schema.GroupResource
+
+	switch r := genericReconciler.(type) {
+	case *GenericDeploymentReconciler:
+		depls, err := r.Lister(src.GetNamespace()).List(sel)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, d := range depls {
+			objs = append(objs, d)
+		}
+
+		gr = appsv1.Resource("deployment")
+
+	case *GenericServiceReconciler:
+		svcs, err := r.Lister(src.GetNamespace()).List(sel)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, s := range svcs {
+			objs = append(objs, s)
+		}
+
+		gr = servingv1.Resource("service")
+	}
+
+	for _, obj := range objs {
+		if metav1.IsControlledBy(obj, src) {
+			return obj, nil
+		}
+	}
```

Comment:
This PR doesn't cover the edge case where multiple objects are owned by the source. Even if I wanted to, I wouldn't know what to do with the extraneous object(s): keep, delete, ...